enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -4468,10 +4468,14 @@ Constant *GradientUtils::GetOrCreateShadowConstant(
         NewOps.push_back(i == 0 ? C : arg->getOperand(i));
       return arg->getWithOperands(NewOps);
     }
+  } else if (auto arg = dyn_cast<GlobalAlias>(oval)) {
+    return GetOrCreateShadowConstant(Logic, TLI, TA, arg->getAliasee(), mode,
+                                     width, AtomicAdd);
   } else if (auto arg = dyn_cast<GlobalVariable>(oval)) {
     if (arg->getName() == "_ZTVN10__cxxabiv120__si_class_type_infoE" ||
         arg->getName() == "_ZTVN10__cxxabiv117__class_type_infoE" ||
-        arg->getName() == "_ZTVN10__cxxabiv121__vmi_class_type_infoE")
+        arg->getName() == "_ZTVN10__cxxabiv121__vmi_class_type_infoE" ||
+        arg->getName().startswith("??_R")) // any of the MS RTTI manglings
```

Comment:
Should all MS mangled globals have this treatment though? What about user defined globals!

Reply:
I didn't like this either but there were too many, and I didn't know how many more there could be. I think this is not a problem because the closest an MS user can get from C/C++ is `?_R...` which has one question mark and comes from `_R`. Besides, such Identifiers are reserved and shouldn't be used anyways.